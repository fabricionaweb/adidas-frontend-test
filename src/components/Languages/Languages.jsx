import React from 'react';
import { useTranslation } from 'react-i18next';
import styles from './Languages.module.css';

export function Languages() {
  const { t, i18n } = useTranslation();

  const changeLanguage = ({ target }) => {
    i18n.changeLanguage(target.value);
  };

  return (
    <div className={styles.wrapper}>
      <div className={styles.container}>
        <select
          data-testid="languages"
          value={i18n.language}
          onChange={changeLanguage}
        >
          <option value="en">{t('English')}</option>
          <option value="de">{t('Germany')}</option>
        </select>
      </div>
    </div>
  );
}

export default Languages;
