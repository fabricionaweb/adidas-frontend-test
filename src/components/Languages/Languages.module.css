.wrapper {
  background-color: #E1E1E1;
  padding: 0.5rem 0;
}

.container {
  max-width: 900px;
  width: 92%;
  margin: 0 auto;
  text-align: right;
}
