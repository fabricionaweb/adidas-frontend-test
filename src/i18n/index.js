import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import resources from './resources';

i18n
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    resources,
    lng: 'en', // default language
    fallbackLng: 'en', // use en if detected lng is not available
    keySeparator: false, // we do not use keys in form messages.welcome
    debug: process.env.NODE_ENV.includes('dev'),
    interpolation: {
      escapeValue: false, // react already safes from xss
    },
  });

export default i18n;
