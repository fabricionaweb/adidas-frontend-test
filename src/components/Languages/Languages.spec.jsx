import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Languages } from './Languages';
import '../../i18n';

describe('Languages.jsx', () => {
  it('should render without crashing', () => {
    const { asFragment } = render(<Languages />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('default language "en"', () => {
    it('should contains "English" option', () => {
      const { getByText } = render(<Languages />);
      expect(getByText('English')).toBeTruthy();
    });

    it('should contains "Germany" option', () => {
      const { getByText } = render(<Languages />);
      expect(getByText('Germany')).toBeTruthy();
    });
  });

  describe('change language to "de"', () => {
    const event = {
      target: { value: 'de' },
    };

    it('should contains "Englisch" option', () => {
      const { getByText, getByTestId } = render(<Languages />);

      fireEvent.change(getByTestId('languages'), event);
      expect(getByText('Englisch')).toBeTruthy();
    });

    it('should contains "Deutschland" option', () => {
      const { getByText, getByTestId } = render(<Languages />);

      fireEvent.change(getByTestId('languages'), event);
      expect(getByText('Deutschland')).toBeTruthy();
    });
  });
});
