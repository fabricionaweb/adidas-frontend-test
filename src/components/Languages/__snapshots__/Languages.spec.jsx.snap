// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Languages.jsx should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="container"
    >
      <select
        data-testid="languages"
      >
        <option
          value="en"
        >
          English
        </option>
        <option
          value="de"
        >
          Germany
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
